### PR TITLE
add e2e for NodeUpgradeJob

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
add e2e for NodeUpgradeJob
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/pkg/edgehub/upgrade/upgrade.go [hunk 1]
```diff
@@ -128,14 +135,27 @@ func (*keadmUpgrade) Upgrade(upgradeReq *commontypes.NodeUpgradeJobRequest) erro
...
 
 	klog.Infof("Begin to run upgrade command")
-	upgradeCmd := fmt.Sprintf("keadm upgrade --upgradeID %s --historyID %s --fromVersion %s --toVersion %s --config %s --image %s > /tmp/keadm.log 2>&1",
-		upgradeReq.UpgradeID, upgradeReq.HistoryID, version.Get(), upgradeReq.Version, opts.ConfigFile, image)
+	upgradeCmd := fmt.Sprintf("keadm upgrade --upgradeID %s --historyID %s --fromVersion %s --toVersion %s --config %s --image %s --binaryPath %s > /tmp/keadm.log 2>&1",
+		upgradeReq.UpgradeID, upgradeReq.HistoryID, version.Get(), upgradeReq.Version, opts.ConfigFile, image, binaryPath)
+
+	// only used in test
+	if upgrade.fake {
+		// add a `--fake=true` flag to indicate that we'll not do actual Upgrade operations
```
Comment (zhu733756) [inline edge/pkg/edgehub/upgrade/upgrade.go:155]: why not use a dry-run flag, I think it more common.

File: tests/e2e/utils/operation.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "NodeUpgradeJob",
+			APIVersion: "operations.kubeedge.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "fake-node-upgrade-job",
+		},
+		Spec: operationsv1alpha1.NodeUpgradeJobSpec{
+			Version:        "v1.12.0",
+			UpgradeTool:    "fake",
```
Comment (zhu733756) [inline tests/e2e/utils/operation.go:97]: Version can be a variable here?

Seems that we can add more cases here?

Comment (gy95) [reply tests/e2e/utils/operation.go:97]: yes, good catch 👍

File: edge/pkg/edgehub/upgrade/upgrade.go (hunk 1 above)
Comment (gy95) [reply edge/pkg/edgehub/upgrade/upgrade.go:155]: ok, will fix this

File: tests/e2e/deployment/operation_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+				}
+
+				return true, nil
+			})
+
+			Expect(pollErr).To(BeNil())
+
+			// delete NodeUpgradeJob
+			err = utils.HandleNodeUpgradeJob(edgeClientSet, http.MethodDelete, expect.Name)
+			Expect(err).Should(BeNil())
```
[the remark this answers is not in the document]
Comment (gy95) [reply tests/e2e/deployment/operation_test.go:95]: I thought AfterEach function is a way to ensure there's no exist cr. i can fix it.

**Pull request closed without merging**